ert/data/_record.py

```diff
@@ -131,34 +137,71 @@ class BlobRecord(Record):
     data: blob_record_data
 
 
-class RecordCollection(_DataElement):
-    records: Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
-    ensemble_size: Optional[int] = None
+_RecordTupleType = Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
+
+
+class RecordCollection:
+    def __init__(self, records: Any, ensemble_size: Optional[PositiveInt] = None):
+        self._records: _RecordTupleType
+        try:
+            # Passing a single records indicates that a uniform record is created.
+            record = self._make_record(records)
+            if ensemble_size is None:
+                raise ValueError("Ensemble size missing for uniform record")
+            self._records = cast(_RecordTupleType, (record,) * ensemble_size)
+            self._ensemble_size = ensemble_size
+            self._is_uniform = True
+        except ValueError:
```

Comment:
This might be a bit fragile. The assumption here is that only we will raise this exception. Also makes the code like a garden path. https://en.wikipedia.org/wiki/Garden-path_sentence

Reply:
Yes, that code is flawed. Will rewrite.